ui/desktop/tests/acceptance/projects/sessions/session-test.js

```diff
@@ -148,13 +148,135 @@ module('Acceptance | projects | sessions | session', function (hooks) {
     await click(TARGET_CONNECT_BUTTON);
 
     assert.strictEqual(currentURL(), urls.session);
-    assert.dom('credential-panel-header').doesNotExist();
-    assert.dom('credential-panel-body').doesNotExist();
+    assert.dom('.credential-panel-header').doesNotExist();
+    assert.dom('.credential-panel-body').doesNotExist();
     assert
       .dom('[data-test-no-credentials]')
       .hasText(`Connected You can now access ${instances.target.name}`);
   });
 
+  test('visiting session with vault type credentials', async function (assert) {
+    this.ipcStub.withArgs('cliExists').returns(true);
+    this.ipcStub.withArgs('connect').returns({
+      session_id: instances.session.id,
+      address: 'a_123',
+      port: 'p_123',
+      protocol: 'tcp',
+      credentials: [
+        {
+          credential_source: {
+            id: 'clvlt_4cvscMTl0N',
+            name: 'Credential Library 0',
+            description: 'Source Description',
+            credential_store_id: 'csvlt_Q1HFGt7Jpm',
+            type: 'vault-generic',
+          },
+          secret: {
+            raw: 'eyJhcnJheSI6WyJvbmUiLCJ0d28iLCJ0aHJlZSIsIm9uZSIsInR3byIsInRocmVlIiwib25lIiwidHdvIiwidGhyZWUiLCJvbmUiLCJ0d28iLCJ0aHJlZSJdLCJuZXN0ZWQiOnsiYm9vbCI6dHJ1ZSwibG9uZyI6IjEyMjM1MzQ1NmFzZWRmYTQzd3J0ZjIzNGYyM2FzZGdmYXNkZnJnYXdzZWZhd3NlZnNkZjQiLCJzZWNlcmV0Ijoic28gbmVzdGVkIn0sInRlc3QiOiJwaHJhc2UifQ',
+            decoded: {
+              data: {
+                backslash: 'password\\withslash\\',
+                backslash1: 'password\\fwithslash\\f',
+                password: '123',
+                username: 'test',
+                email: {
+                  address: 'test.com',
+                },
+              },
+              metadata: {
+                created_time: '2024-04-12T18:38:36.226715555Z',
+                custom_metadata: null,
+                deletion_time: '',
+                destroyed: false,
+                version: 8,
+              },
+            },
+          },
+        },
+      ],
+    });
+
+    await visit(urls.target);
+    await click(TARGET_CONNECT_BUTTON);
+    assert.strictEqual(currentURL(), urls.session);
+
+    assert.dom('.credential-panel-header').exists();
+    assert.dom('.credential-panel-body').exists();
+    assert.dom('.credential-secret').exists();
+
+    assert
+      .dom('.secret-container:nth-of-type(1)')
+      .hasText('username ■ ■ ■ ■ ■ ■ ■ ■ ■ ■ ■ ■');
```

Comment:
None blocking comment: not sure what is the purpose of testing this.
As per my understanding, we are asserting the secret is by default "hidden". IMHO that's a component test that [we already do here](https://github.com/hashicorp/boundary-ui/blob/main/ui/desktop/tests/integration/components/hidden-secret-test.js#L40-L44).

Reply:
yeah, the component handles that part, I refactored the test to only cover nested and esc chars which are covered in this PR. 

I added this originally as there were no tests covering a session with a credential. We only had one test case that covered "visiting session with no credential" but there was nothing to test the case, "visiting session with a credential". 

After some thinking, I refactored this to two new test cases that are addressed in this PR and previous PR related to this [work](https://hashicorp.atlassian.net/browse/FRB-192) 

1. visiting a session with a static credential (handle nested without esc chars)
2. visiting a session with a vault credential (handle nested without esc chars)